modules/frontend/metrics_query_handler.go

```diff
@@ -0,0 +1,171 @@
+package frontend
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/grafana/dskit/user"
+	"github.com/grafana/tempo/modules/frontend/combiner"
+	"github.com/grafana/tempo/modules/frontend/pipeline"
+	"github.com/grafana/tempo/pkg/api"
+	"github.com/grafana/tempo/pkg/tempopb"
+)
+
+// newMetricsQueryInstantHTTPHandler handles instant queries.  Internally these are rewritten as query_range with single step
+// to make use of the existing pipeline.
+func newMetricsQueryInstantHTTPHandler(cfg Config, next pipeline.AsyncRoundTripper[combiner.PipelineResponse], logger log.Logger) http.RoundTripper {
```

Comment:
can we add a streaming version? is there something that makes it difficult?

Reply:
I left it out because there was still ongoing work in that area: https://github.com/grafana/tempo/pull/3808   If we are good, yep I can add it.

> is there something that makes it difficult

I should be asking you that 😄  No I don't think so, it should be nearly identical to query_range streaming.

Reply:
> I should be asking you that 😄 No I don't think so, it should be nearly identical to query_range streaming.

I didn't think so either, but thought i might be missing something :). Agree it would basically identical to query_range streaming.

Reply:
Added.  Nice, it's quite fun to stream `{ } | count_over_time()` and watch the value count up.